projects/js-toolkit/packages/js-toolkit-core/src/project/liferayCli/Dist.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * SPDX-FileCopyrightText: © 2021 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+import FilePath from '../../file/FilePath';
+import LiferayJson from '../../schema/LiferayJson';
+import Project from './Project';
+
+export default class Dist {
+	readonly dir: FilePath | null;
+	readonly file: FilePath | null;
+
+	constructor(_project: Project, liferayJson: LiferayJson) {
+		switch (liferayJson.build.type) {
+			case 'customElement':
+				this.dir = null;
+				this.file = null;
+				break;
+
+			case 'bundler2':
+			default: {
+				/* eslint-disable-next-line @typescript-eslint/no-var-requires */
+				const bundler2Project = require('liferay-npm-build-tools-common/lib/project');
+
+				this.dir = new FilePath(
+					bundler2Project.jar.outputDir.asNative
+				).resolve();
+				this.file = this.dir.join(bundler2Project.jar.outputFilename);
+				break;
```

Comment:
That was to account for the case where the build type is not configured (for backwards compatibility). However, because we have the `Project` class that simulates configuration defaults when they are not present, I think I can refactor this and make it behave as you say, which is much cleaner :+1: